packages/nocodb/src/db/BaseModelSqlv2.ts

```diff
@@ -1509,162 +1509,196 @@ class BaseModelSqlv2 {
-              };
-            } else if (colOptions.type === 'bt') {
-              // @ts-ignore
-              const colOptions =
-                (await column.getColOptions()) as LinkToAnotherRecordColumn;
-              const pCol = await Column.get({
-                colId: colOptions.fk_parent_column_id,
-              });
-              const cCol = await Column.get({
-                colId: colOptions.fk_child_column_id,
-              });
-              const readLoader = new DataLoader(async (ids: string[]) => {
-                const data = await (
-                  await Model.getBaseModelSQL({
-                    id: pCol.fk_model_id,
-                    dbDriver: this.dbDriver,
-                  })
-                ).list(
-                  {
-                    // limit: ids.length,
-                    where: `(${pCol.column_name},in,${ids.join(',')})`,
-                    fieldsSet: (readLoader as any).args?.fieldsSet,
-                  },
-                  true,
-                );
-                const gs = groupBy(data, pCol.title);
-                return ids.map(async (id: string) => gs?.[id]?.[0]);
-              });
+                    return data;
+                  } else {
+                    return [
+                      await this.mmList(
+                        {
+                          parentId: ids[0],
+                          colId: column.id,
+                        },
+                        (listLoader as any).args,
+                      ),
+                    ];
+                  }
+                });
 
-              // defining HasMany count method within GQL Type class
-              proto[column.title] = async function (args?: any) {
-                if (
-                  this?.[cCol?.title] === null ||
-                  this?.[cCol?.title] === undefined
-                )
-                  return null;
+                const self: BaseModelSqlv2 = this;
+                // const childColumn = await colOptions.getChildColumn();
+                proto[
+                  column.uidt === UITypes.Links
+                    ? `_nc_lk_${column.title}`
+                    : column.title
+                ] = async function (args): Promise<any> {
+                  (listLoader as any).args = args;
+                  return await listLoader.load(
+                    getCompositePk(self.model.primaryKeys, this),
+                  );
+                };
+              } else if (colOptions.type === 'bt') {
+                // @ts-ignore
+                const colOptions =
+                  (await column.getColOptions()) as LinkToAnotherRecordColumn;
+                const pCol = await Column.get({
+                  colId: colOptions.fk_parent_column_id,
+                });
+                const cCol = await Column.get({
+                  colId: colOptions.fk_child_column_id,
+                });
+                const readLoader = new DataLoader(async (_ids: string[]) => {
```

Comment:
here i think you can add some examples to show how the transformation is done. more easy to visualise and understand.